### PR TITLE
Chore: Add localization regression tests to prevent format string crashes

### PR DESCRIPTION
**Pull request by SkinnnyJay**
## Summary

Add comprehensive regression tests to prevent localization format string crashes and ensure translation completeness. These tests complement the recently merged Swift String Catalogs migration ([#691](https://github.com/permissionlesstech/bitchat/pull/691)) by providing runtime safety validation.

The failed tests should be fixed once https://github.com/permissionlesstech/bitchat/pull/699 is merged to main

## Changes

- Add 7 regression tests for pluralized format strings (`%#@variable@` syntax)
- Test edge cases: large numbers, zero values, empty strings
- Validate format string argument matching to prevent crashes
- Ensure `String(localized:)` calls don't throw runtime exceptions
- Maintain existing structural validation tests (locale parity, placeholder consistency)

## Testing

- All tests build successfully with `swift build`
- Tests prevent crashes from malformed localization calls
- Validates Korean translations alongside existing 14 languages
- Covers critical UI strings: people count, location notes, channel titles

## Impact

Prevents production crashes from localization format string mismatches while maintaining the comprehensive translation coverage established in the String Catalogs migration.

### CODE REVIEW

**Comment by SkinnnyJay** [comment]
@codex review

**Comment by SkinnnyJay** [comment]
This should be fixed once https://github.com/permissionlesstech/bitchat/pull/699 is merged to `main`

File: bitchatTests/Localization/LocalizationCatalogTests.swift [hunk 1]
```diff
@@ -239,6 +239,151 @@ final class LocalizationCatalogTests: XCTestCase {
...
+    }(), "Location notes header localization should not throw")
+  }
+
+  func testFormatStringArgumentMatching() {
+    // Verify that the format strings properly handle their expected arguments
+    
+    // People count expects: %#@people@ format with Int
+    let peopleResult = String(
+      localized: "content.accessibility.people_count",
+      comment: "Test"
```
[the remark this answers is not in the document]
Comment (SkinnnyJay) [reply bitchatTests/Localization/LocalizationCatalogTests.swift:292]: **Exact Issues Addressed:**
- ✅ "call `String(localized:)` from the test target without specifying a bundle" → Now all calls specify `bundle: appBundle()`
- ✅ "simply returns the key" → Added assertions to verify we get translations, not keys
- ✅ "never touches the actual pluralized strings" → Now accesses real app's `Localizable.xcstrings`
- ✅ "provide no protection against the regression" → Tests now catch real localization issues
